zerver/lib/actions.py

```diff
@@ -2299,6 +2295,11 @@ def check_message(sender: UserProfile, client: Client, addressee: Addressee,
     if realm is None:
         realm = sender.realm
 
+    mention_data = MentionData(
+        realm_id=realm.id,
+        content=message_content
+    )
+
     if addressee.is_stream():
```

Comment:
I have made this change to extract `MentionData` in `check_message` instead of `do_send_messages` because I think that it would be better to check for the message here itself instead of doing it in `do_send_messages`, though this leads to some more changes like in management commands where we send messages directly calling `do_send_messages` and not call `check_message`, (Just mentioning it as I am not fully sure that whether this is the best approach or not.)